test/e2e/multiple_worker_node_groups_test.go

```diff
@@ -52,6 +52,46 @@ func TestVSphereKubernetes124BottlerocketAndAndRemoveWorkerNodeGroups(t *testing
 	)
 }
 
+func TestVSphereKubernetes124BottlerocketUpgradeAndRemoveWorkerNodeGroupsAPI(t *testing.T) {
```

Comment:
Does this test setup a self-managed cluster right?  I don't think we can use self-managed clusters for the API tests since it does not support it.

Can we use `MulticlusterE2ETest` to setup the test instead? Check the `TestVSphereMulticlusterWorkloadClusterAPI` for an example.